diff --git a/src/Entity/User.php b/src/Entity/User.php
index a6ef78c..1731e8e 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -3,11 +3,13 @@
 namespace App\Entity;
 
 use App\Repository\UserRepository;
+use Database\SoftDelete\Core\Attribute\SoftDeleteUniqueIndex;
 use Database\SoftDelete\Core\Contract\SoftDeletableInterface;
 use Doctrine\ORM\Mapping as ORM;
 
 #[ORM\Entity(repositoryClass: UserRepository::class)]
 #[ORM\Table(name: '`user`')]
+#[SoftDeleteUniqueIndex(fields: ['email'])]
 class User implements SoftDeletableInterface
 {
     #[ORM\Id]
